types/selenium-webdriver/lib/select.d.ts

```diff
@@ -0,0 +1,112 @@
+import { WebElement } from '../';
+
+export class Select {
+    /**
+     * Create an Select Element
+     * @param {WebElement} element Select WebElement.
+     */
+    constructor(element: WebElement);
+
+    element: WebElement;
+
+    /**
+     * Select the option at the given index. This is done by examining the "index" attribute of an
+     * element, and not merely by counting.
+     *
+     * @param {Number} index The option at this index will be selected
+     * @return {Promise<void>}
+     */
+    selectByIndex(index: number): Promise<void>;
+
+    /**
+     *
+     * Select option by specific value.
+     *
+     * <example>
+       <select id="selectbox">
+       <option value="1">Option 1</option>
+       <option value="2">Option 2</option>
+       <option value="3">Option 3</option>
+       </select>
+       const selectBox = await driver.findElement(By.id("selectbox"));
+       await selectObject.selectByValue("Value");
+     * </example>
+     *
+     *
+     * @param {string} value value of option element to be selected
+     */
+    selectByValue(value: string): Promise<void>;
+
+    /**
+     *
+     * Select option with displayed text matching the argument.
+     *
+     * <example>
+       <select id="selectbox">
+       <option value="1">Option 1</option>
+       <option value="2">Option 2</option>
+       <option value="3">Option 3</option>
+       </select>
+       const selectBox = await driver.findElement(By.id("selectbox"));
+       await selectObject.selectByVisibleText("Option 2");
+     * </example>
+     *
+     * @param {String|Number} text       text of option element to get selected
+     *
+     */
+    selectByVisibleText(text: string|number): Promise<void>;
+
+    /**
+     * @return {!Promise<!Array<!WebElement>>} All options belonging to this select tag
+     */
+    getOptions(): Promise<WebElement[]>;
+
+    /**
+     * Retruns a boolean value if the select tag is multiple
+     * @returns {Promise<boolean>}
+     */
+    isMultiple(): Promise<boolean>;
+
+    /**
+     * Returns a list of all selected options belonging to this select tag
+     *
+     * @returns {Promise<void>}
+     */
+    getAllSelectedOptions(): Promise<WebElement[]>;
+
+    /**
+     * Returns first Selected Option
+     * @returns {Promise<Element>}
+     */
+    getFirstSelectedOption(): Promise<WebElement>;
```

Comment:
```suggestion
    getFirstSelectedOption(): Promise<WebElement | undefined>;
```
Looking at [the implementation](https://github.com/SeleniumHQ/selenium/blob/trunk/javascript/node/selenium-webdriver/lib/select.js#L321-L323), this method returns `undefined` when there are no selected options (since element 0 of an empty array is `undefined`).

Reply:
Thank you @SupernaviX. Changes committed.